### DISCUSSION TITLE
Using .env?

### DISCUSSION THREAD

**Discussion by ThatXliner** [jdx/mise#3068] [Q&A]
I'm coming from asdf and direnv.

I've always abused (and under-used) direnv as just a tool to load my .env ([and `.env.local`](https://github.com/direnv/direnv/commit/7ecfe62cee70e3a4a98203c2ef96c132f95f589b) on [my custom fork](https://github.com/ThatXliner/direnv)) files as simple containers for environment variables. I don't like `mise.toml` (as it's not commonly used), and I would like it if mise simply just loaded my `.env*` files when I change directory to it.

Sure, I could just hack something up in my shell and not complain here but I was wondering why mise doesn't do this? What's the rationale/philosophy for not supporting such a use case? Or like, the direnv plugin that I installed didn't function correctly (when I `cd` out, the environment variables are still loaded, at least according to [my prompt](https://github.com/romkatv/powerlevel10k))

**Comment by jdx** [comment]
we do have support for this

**Reply by ThatXliner** [reply]
is there a way to specify multiple files for [`MISE_ENV_FILE`](https://mise.jdx.dev/configuration.html#mise-env-file)?

**Reply by Marukome0743** [reply]
Multiple dotenv files are supported through the `env._.file` directive:

```toml
[env]
_.file = [".env", ".env.local"]
```

I verified this against the current mise main branch: both files are loaded, and when they define the same variable, the file listed later (`.env.local` in this example) takes precedence.

This is the recommended approach when multiple dotenv files are needed. `MISE_ENV_FILE` itself is not the multi-file configuration mechanism; the array form of `_.file` should be used instead.

The supported `_.file` formats and multiple-file example are also documented in the mise environment documentation.

*AI-assisted — Tool: Codex; model: OpenAI/GPT-5; version: unavailable.*
